galley/pkg/config/resource/name.go

```diff
@@ -28,7 +28,12 @@ func NewName(namespace, local string) Name {
 	return Name{string: namespace + "/" + local}
 }
 
-// String inteface implementation.
+// NewFullName returns a given name as a resource Name
+func NewFullName(name string) Name {
```

Comment:
@ozevren is there an existing way to create a full name from a string?

Reply:
https://github.com/istio/istio/blob/3b30a934454186de347585d2b287ff744cbf9d35/galley/pkg/config/resource/name.go#L23

Reply:
There used to be one, in one of the previous incarnations, but I believe it is gone now as it wasn't used. It should be fairly easy to add one here.

Reply:
This should validate the name before passing it blindly. For example an empty name, or a name with an empty namespace segment should not be valid.

Reply:
@ozevren what should be the behaviour here, if the name is empty or does not contain namespace should it return an error? because I can't see anything else can be done here.